source/node/myButtercup/MyButtercupClient.js

```diff
@@ -1,202 +1,493 @@
+    OAUTH_AUTHORISE_URI,
+    OAUTH_REDIRECT_URI,
+    OAUTH_TOKEN_URI
+} = require("./symbols.js");
+const { ensureVaultContentsEncrypted } = require("../tools/validation.js");
 
-const NOOP = () => {};
+/**
+ * @typedef {Object} MyButtercupShareBase
+ * @property {String} id The share ID
+ * @property {String} title The share title
+ * @property {Boolean} perm_read Permission to read
+ * @property {Boolean} perm_write Permission to write changes
+ * @property {Boolean} perm_manage Permission to share with others, remove share access etc.
+ */
 
-let __shared;
+/**
+ * @typedef {MyButtercupShareBase} MyButtercupIncomingShare
+ * @property {String} share_password_enc Encrypted password for the share
+ * @property {Number} sharing_user_id The user that shared the item
+ * @property {String} sharing_user_key The public key of the user for the share (used
+ *  for decrypting the share password)
+ */
 
-function handleGetResponse(res) {
-    if (res.status < 200 || res.status >= 300) {
-        throw new Error(`Invalid API response status: ${res.status} ${res.statusText}`);
+/**
+ * @typedef {MyButtercupShareBase} MyButtercupEncryptedShare
+ * @property {String} content Encrypted share content
+ */
+
+/**
+ * @typedef {Object} MyButtercupOrganisation
+ * @property {Number} id The organisation's ID
+ * @property {String} name The organisation name
+ * @property {String} created The creation date
+ */
+
+/**
+ * @typedef {Object} MyButtercupDigest
+ * @property {Number} archive_id The ID of the user's archive
+ * @property {String} public_key The RSA public key for the user
+ * @property {Array.<Object>} messages System messages for the user (internal processing)
+ * @property {Array.<MyButtercupIncomingShare>} new_shares An array of new shares to process
+ * @property {Array.<MyButtercupOrganisation>} organisations An array of user organisations
+ */
+
+/**
+ * @typedef {Object} MyButtercupUsersListItem
+ * @property {Number} user_id The ID of the user
+ * @property {Number} organisation_id The organisation ID the user was found in
+ * @property {String} name The name of the user
+ * @property {String} public_key The public key for the user
+ */
+
+/**
+ * @typedef {Object} MyButtercupTokenResult
+ * @property {String} accessToken An OAuth2 access token for API requests
+ * @property {String} refreshToken An OAuth2 refresh token
+ */
+
+/**
+ * @typedef {Object} MyButtercupArchiveDetails
+ * @property {Number} id The remote vault ID
+ * @property {Number} updateID The current update ID for the vault
+ * @property {String} created The creation date
+ * @property {String} lastUpdate The last update date
+ */
+
+function demultiplexShares(sharesTxt) {
+    const shares = {};
+    const lines = sharesTxt.split("\n");
+    while (lines.length > 0) {
+        const propLine = lines.shift();
+        if (!/^\<--\(/.test(propLine)) {
+            continue;
+        }
+        const payload = JSON.parse(propLine.replace(/^\<--\(/, "").replace(/\)--\>$/, ""));
```

Comment:
Should this be in `try/catch` just in case?

Reply:
I wrapped it in one now, but of course it's a non-recoverable error if this does fail..